skimage/measure/_marching_cubes_lewiner_cy.pyx

```diff
@@ -973,27 +975,49 @@ def marching_cubes(float [:, :, :] im not None, double isovalue, LutProvider lut
             while x < Nx_bound:
                 x += st
                 x_st = x + st
-
-                # Initialize cell
-                cell.set_cube(isovalue, x, y, z, st,
-                    im[z   ,y, x], im[z   ,y, x_st], im[z   ,y_st, x_st], im[z   ,y_st, x],
-                    im[z_st,y, x], im[z_st,y, x_st], im[z_st,y_st, x_st], im[z_st,y_st, x] )
-
-                # Do classic!
-                if classic:
-                    # Determine number of vertices
-                    nt = 0
-                    while luts.CASESCLASSIC.get2(cell.index, 3*nt) != -1:
-                        nt += 1
-                    # Add triangles
-                    if nt > 0:
-                        cell.add_triangles(luts.CASESCLASSIC, cell.index, nt)
-                else:
-                    # Get case, if non-nul, enter the big switch
-                    case = luts.CASES.get2(cell.index, 0)
-                    if case > 0:
-                        config = luts.CASES.get2(cell.index, 1)
-                        the_big_switch(luts, cell, case, config)
+                if mask is None:
+                    # Initialize cell
+                    cell.set_cube(isovalue, x, y, z, st,
+                        im[z   ,y, x], im[z   ,y, x_st], im[z   ,y_st, x_st], im[z   ,y_st, x],
+                        im[z_st,y, x], im[z_st,y, x_st], im[z_st,y_st, x_st], im[z_st,y_st, x] )
+
+                    # Do classic!
+                    if classic:
+                        # Determine number of vertices
+                        nt = 0
+                        while luts.CASESCLASSIC.get2(cell.index, 3*nt) != -1:
+                            nt += 1
+                        # Add triangles
+                        if nt > 0:
+                            cell.add_triangles(luts.CASESCLASSIC, cell.index, nt)
+                    else:
+                        # Get case, if non-nul, enter the big switch
+                        case = luts.CASES.get2(cell.index, 0)
+                        if case > 0:
+                            config = luts.CASES.get2(cell.index, 1)
+                            the_big_switch(luts, cell, case, config)
+
+                elif mask[z_st, y_st, x_st]:
```

Comment:
I see in [`find_contours`](https://github.com/scikit-image/scikit-image/blob/v0.16.1/skimage/measure/_find_contours_cy.pyx#L103) that the neighborhood of the points are also checked for not being masked. Shouldn't this be also done here?

Reply:
okay after testing here https://github.com/Leguark/Sandbox/blob/master/Masked%20marching%20cubes.ipynb

I do not see necessary that condition. I guess the behaviour changes a bit. Since marching cubes acts on the edges, masking a voxel is a bit ambiguous if the edge should be masked or not. For my taste is better without the condition.

Reply:
In a `if A or B` test, **B** is not evaluated if **A** is true. To illustrate this, see this snippet:
```python
>>> if True or 1/0: print('yes')
... 
yes
```
This will not cause any overhead and avoid code duplication.

It's OK to not check voxels neighbors, It was just a small remark ;-)

Reply:
That is good information, then I am going to add your suggestion